sycl/plugins/hip/pi_hip.cpp

```diff
@@ -2131,8 +2131,7 @@ pi_result hip_piMemGetInfo(pi_mem memObj, pi_mem_info queriedInfo,
 /// \return PI_SUCCESS
 pi_result hip_piextMemGetNativeHandle(pi_mem mem,
                                       pi_native_handle *nativeHandle) {
-  *nativeHandle =
-      reinterpret_cast<pi_native_handle>(mem->mem_.buffer_mem_.get());
+  *nativeHandle = static_cast<pi_native_handle>(mem->mem_.buffer_mem_.get());
```

Comment:
From the issue:

> Replacing the reinterpret_cast by static_cast avoids the compilation error, but might be unsafe due to narrowing.

Can we have a check that upper bits are zero and return some failure code if that's not the case?

Reply:
Hi @aelovikov-intel, thanks for the review.

I've added a check to make sure that the upper bits are zero.